ports/rp2/modmachine.c

```diff
@@ -80,10 +82,19 @@ STATIC mp_obj_t machine_bootloader(void) {
 }
 MP_DEFINE_CONST_FUN_OBJ_0(machine_bootloader_obj, machine_bootloader);
 
-STATIC mp_obj_t machine_freq(void) {
-    return MP_OBJ_NEW_SMALL_INT(clock_get_hz(clk_sys));
+STATIC mp_obj_t machine_freq(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return MP_OBJ_NEW_SMALL_INT(clock_get_hz(clk_sys));
+    } else {
+        mp_int_t freq = mp_obj_get_int(args[0]);
+        if (set_sys_clock_khz(freq / 1000, false)) {
+            return mp_const_none;
+        } else {
+            mp_raise_ValueError(MP_ERROR_TEXT("valid range 12-270MHz"));
```

Comment:
I get that this is nor completely arbitrary, on the other hand there doesn't seem to be a guarantee that these numbers actually work, nor that numbers out if this range won't, so seems better to just state "cannot change frequency"?

Reply:
My first message was something like "invalid frequency". But then I thought could give people a hint, after trying a handful values. But testing now the range for the full MHz values, 45 values of that range are not accepted. So a more vague text is better. I'll change that. Since it's a ValueError and the call has just a single argument, it might not need a message at all. Because that's what it is.

Reply:
All odd numbers i.e. clock divider cannot do that? That mechanism is probably documented somewhere..

Reply:
Not all odd numbers. 141, 147, 153, 159, 165,  ... are OK, and everything between 12 and 136 is fine too, as well as 10.